nixos/modules/services/networking/netbird-server.nix

```diff
@@ -0,0 +1,529 @@
+          envsubst '$NETBIRD_MGMT_API_ENDPOINT' < "$OIDC_TRUSTED_DOMAINS".tmpl > "$OIDC_TRUSTED_DOMAINS"
+        '';
+      };
+
+      systemd.services.netbird-signal = {
+        after = [ "network.target" ];
+        wantedBy = [ "netbird-management.service" ];
+        restartTriggers = [ settingsFile managementFile ];
+
+        serviceConfig = {
+          ExecStart = ''
+            ${cfg.package}/bin/netbird-signal run \
+              --port ${builtins.toString cfg.ports.signal} \
+              --log-file console \
+              --log-level ${cfg.logLevel}
+          '';
+          Restart = "always";
+          RuntimeDirectory = "netbird-mgmt";
+          StateDirectory = "netbird-mgmt";
+          WorkingDirectory = stateDir;
+        };
+        unitConfig = {
+          StartLimitInterval = 5;
+          StartLimitBurst = 10;
+        };
+        stopIfChanged = false;
+      };
+
+      systemd.services.netbird-management = {
+        description = "The management server for Netbird, a wireguard VPN";
+        documentation = [ "https://netbird.io/docs/" ];
+        after = [ "network.target" "netbird-setup.service" ];
+        wantedBy = [ "multi-user.target" ];
+        wants = [ "netbird-signal.service" "netbird-setup.service" ];
+        restartTriggers = [ settingsFile managementFile ];
+
+        serviceConfig = {
+          ExecStart = ''
+            ${cfg.package}/bin/netbird-mgmt management \
+              --config ${stateDir}/management.json \
+              --datadir ${stateDir}/data \
+              ${optionalString cfg.management.disableAnonymousMetrics "--disable-anonymous-metrics"} \
+              ${optionalString cfg.management.disableSingleAccountMode "--disable-single-account-mode"} \
+              --dns-domain ${cfg.management.dnsDomain} \
+              --single-account-mode-domain ${cfg.management.singleAccountModeDomain} \
+              --idp-sign-key-refresh-enabled \
+              --port ${builtins.toString cfg.ports.management} \
+              --log-file console \
+              --log-level ${cfg.logLevel}
+          '';
+          Restart = "always";
+          RuntimeDirectory = "netbird-mgmt";
+          StateDirectory = [ "netbird-mgmt" "netbird-mgmt/data" ];
+          WorkingDirectory = stateDir;
+        };
+        unitConfig = {
+          StartLimitInterval = 5;
+          StartLimitBurst = 10;
+        };
+        stopIfChanged = false;
+      };
+    })
+
+    (mkIf cfg.enableCoturn {
+      services.coturn = {
+        enable = true;
+
+        realm = settings.NETBIRD_DOMAIN;
+        lt-cred-mech = true;
+        no-cli = true;
+
+        extraConfig = ''
+          fingerprint
+
+          user=${settings.TURN_USER}:${builtins.toString settings.TURN_PASSWORD}
```

Comment:
As I said earlier, it is possible to pass this via the command like. Also the config should probably set `external-ip` to `NETBIRD_DOMAIN`

Reply:
`external-ip` expects an ip address, which is not what `NETBIRD_DOMAIN` contains

Reply:
Still works as expected though. I tested this with coturn behind two layers of NAT